src/megaapi_impl.cpp

```diff
@@ -7489,6 +7490,30 @@ bool MegaApiImpl::hasToForceUpload(const Node &node, const MegaTransferPrivate &
     return canForceUpload && (isMedia || isPdf) && !(hasPreview && hasThumbnail);
 }
 
+void MegaApiImpl::platformSetRLimitNumFile(int newNumFileLimit) const
+{
+#ifndef WIN32
+    struct rlimit rl{0,0};
+    if (0 < getrlimit(RLIMIT_NOFILE, &rl))
+    {
+        auto e = errno;
+        LOG_err << "Error calling getrlimit: " << e;
+    }
+    else
+    {
+        LOG_info << "rlimit for NOFILE before change is: " << rl.rlim_cur << ", " << rl.rlim_max;
+        rl.rlim_cur = rlim_t(newNumFileLimit);
```

Comment:
Should we avoid the call `if (newNumFileLimit > rl.rlim_max)`?? maybe not worthy if it simply fails without side-effects, falling into the error's block below.

Reply:
Or maybe we could force `rl.rlim_max` if the requested limit is above the actual limit in the platform :thinking:

Reply:
Yes, we'll log an error if the OS won't accept the change.   And we're not allowed to change the max, anyway (except with admin privilege).  On iOS that max number is extremely large anyway, it might be max int.

Reply:
In Android, the max number depends on the device/Android version. In iOS it may change also with the OS version. That's why I suggest the override of the maximum value.